src/Layouts/stars/starsPage/addStar.tsx

```diff
@@ -0,0 +1,183 @@
+import { useState } from 'react';
+import {
+  Dialog,
+  DialogActions,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  TextField,
+  Button,
+  FormControl,
+  InputLabel,
+  Select,
+  Input,
+  MenuItem,
+  Divider,
+} from '@mui/material';
+import {
+  starType,
+  defaultStar,
+  statuses,
+  assignees,
+  severities,
+  versions,
+  computers,
+} from '../../../assets/star';
+import '../styles/stars.css';
+
+interface starProps {
+  isOpen: boolean;
+  toggleModal: (param: boolean) => void;
+  addStar: (star: starType) => void;
+}
+
+const AddStar = ({ isOpen, toggleModal, addStar }: starProps) => {
+  const [newStar, setNewStar] = useState<starType>(defaultStar);
+
+  const setAttr = (attr: keyof starType, value: string | number) => {
+    const s = newStar;
+    newStar.id = Math.random();
+    s[attr] = value as never;
```

Comment:
"the never type represents the type of values that never occur. Variables also acquire the type never when narrowed by any type guards that can never be true."

i didn't quite understand why you used it

Reply:
because of the following error:
Type 'string | number' is not assignable to type 'never'.
  Type 'string' is not assignable to type 'never'.

stackoverflow did not suggest solutions other than "as never"

Reply:
you can use
    `  setNewStar  (Object.assign(newStar, { [attr]: value }));`
    or
`    setNewStar({ ...newStar, ...{ [attr]: value } });`

Reply:
fixed